apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -408,6 +410,7 @@ private void mergeBatch(long maxSeq, Collection<RdaChange<TClaim>> changes) {
                 var metaData = createMetaData(change);
                 entityManager.merge(metaData);
                 entityManager.merge(change.getClaim());
+                samhsaUtil.processClaim(change.getClaim(), entityManager);
```

Comment:
I'm being slightly paranoid here, but we may want to consider doing these inserts in an explicit transaction. Theoretically, if a request comes in right when we're inserting the data, they could get the new claims after they're inserted, but before the SAMHSA tags are created, resulting in a leak. If we do it in a transaction, they shouldn't be able to see the new data until after all the inserts have completed.

Reply:
After a little bit of research, it appears that `executeProcedure` does run in a transaction:

```
Runs a transaction that does not produce a result (usually an insert or update). Ensures that the transaction is properly committed if the procedure completes normally or rolled back if the procedure throws an exception. Internally uses the adapter method ThrowingConsumer.executeAsFunction and discards the null result.
```